main/transaction/index.ts

```diff
@@ -22,19 +22,44 @@ export interface TransactionData extends JsonTx {
   maxFee: string
 }
 
+export interface SignerCompatibility  {
+  signer: string,
+  tx: string,
+  compatible: boolean
+}
+
 function toBN (hexStr: string) {
   return new BN(stripHexPrefix(hexStr), 'hex')
 }
 
-async function populate (rawTx: RawTransaction, signerType: string, chainConfig: Common, gas: any): Promise<TransactionData> {
+function signerCompatibility (txData: TransactionData, signer: string): SignerCompatibility {
+  return txData.type === '0x2' ? ({
+    signer, tx: 'london', compatible: londonHardforkSigners.includes(signer)
+  }) : ({ 
+    signer, tx: 'legacy', compatible: true 
+  })
+}
+
+function londonToLegacy (txData: TransactionData): TransactionData {
+  if (txData.type === '0x2') {
```

Comment:
I think it can be better to return a new object rather than modify the one that was passed in, just to be safe
```suggestion
  if (txData.type === '0x2') {
     const { type, maxFeePerGas, maxPriorityFeePerGas, ...tx } = txData
     return {
        ...tx,
        gasPrice: maxFeePerGas
      }
 }
 
 return txData
```

Reply:
Do we want to keep `maxFeePerGas` `maxPriorityFeePerGas` on the tx object?